cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -207,12 +207,16 @@ void SparsePoseGraph::ComputeConstraintsForOldScans(
   const auto& node_data = optimization_problem_.node_data();
   for (size_t trajectory_id = 0; trajectory_id != node_data.size();
        ++trajectory_id) {
-    for (size_t node_index = 0; node_index != node_data[trajectory_id].size();
-         ++node_index) {
+    for (size_t trimmed_node_index = 0;
```

Comment:
`trimmed_node_index` -> `node_data_index`? Otherwise a bit confusing since this is the index into the untrimmed nodes.